docs/ingestion/concurrent-append-replace.md

```diff
@@ -0,0 +1,145 @@
+---
+id: concurrent-append-replace
+title: Concurrent append and replace
+---
+
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~   http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+:::info
+Concurrent append and replace is an [experimental feature](../development/experimental.md) available for JSON-based batch and streaming. It is not currently available for SQL-based ingestion.
+:::
+
+This feature allows you to safely replace the existing data in an interval of a datasource while new data is being appended to that interval. One of the most common applications of this is appending new data (using say streaming ingestion) to an interval while compaction of that interval is already in progress. 
+
+To set up concurrent append and replace, you need to use the context flag `useConcurrentLocks`. Druid will then determine the correct lock type for you, either append or replace. Although can set the type of lock manually, we don't recommend it. 
```

Comment:
Would it be worth including a description of how the lock types fit into this feature? For example,

>Druid uses lock types to manage data consistency when two tasks are working on the same segment. The concurrent append and replace feature automatically determines the correct lock type to use when ingesting and replacing data at the same time. 

Something like the above, but not sure if my understanding is correct.

Reply:
Another thing I'm not certain on is what part of the feature is concurrent append and replace. Do you need to set `useConcurrentLocks` to enable concurrent and replace (what's currently documented), or is it just a flag to help manage concurrent append and replace? The second sentence kind of makes it sound like you can still use concurrent append/replace but with manual task locks.